### ISSUE TITLE
Self-signed certificates

### ISSUE THREAD

**Issue by kyrofa** [alchemy-charmers/layer-haproxy#18]
I love the Let's Encrypt integration, but it would also be handy if the charm supported generating (or at least using) self-signed certificates for environments where Let's Encrypt isn't suitable.

**Comment by chris-sanders** [comment]
I agree, especially for testing this would be an interesting option. The rub has just been time to implement vs value. I find that many of the integrations that have had me testing this refuse unknown certificates meaning I have to tests against a non-TLS or an actually established certificate anyway.

It's not something I would avoid adding, just something that hasn't been high enough priority to put time on. If anyone wants to take a stab at implementing we would be happy to review it. Otherwise, I'll leave this open it might make it to the top of my queue one day :)
